va/va_fei_hevc.h

```diff
@@ -0,0 +1,282 @@
+ */
+
+/**
+ * \file va_fei_hevc.h
+ * \brief The FEI encoding HEVC special API
+ */
+
+
+#ifndef  __VA_FEI_HEVC_H__
+#define  __VA_FEI_HEVC_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include "va_fei.h"
+
+/** \brief FEI frame level control buffer for Hevc */
+typedef struct _VAEncMiscParameterFEIFrameControlHevc
+{
+    uint32_t      function; /* one of the VAConfigAttribFEIFunctionType values */
+    /** \brief CTB control input buffer. It is valid only when per_ctb_input
+     * is set to 1. The data in this buffer correspond to the input source. CTB is in raster scan order,
+     * each CTB control data structure is defined by VAEncFEICTBControlHevc.
+     * Buffer size shall not be less than the number of CTBs multiplied by
+     * sizeof(VAEncFEICTBControlHevc).
+     */
+    VABufferID    ctb_ctrl;
+    /** \brief CTB cmd per CTB data output of ENC.
+     * Each CTB block has one CTB cmd data with layout VAEncFEICTBCmdHevc
+     * Buffer size shall not be less than the number of CTBs multiplied by
+     * sizeof(VAEncFEICTBCmdHevc).
+     */
+    VABufferID    ctb_cmd;
+    /** \brief CU record data output of ENC.
+     * Each CTB block has one CU record data with layout VAEncFEICURecordLongHevc for SKL and VAEncFEICURecordHevc for ICL+
+     * Buffer size shall not be less than the number of CTB blocks multiplied by
+     * sizeof(VAEncFEICURecordLongHevc) or sizeof(VAEncFEICURecordHevc).
+     */
+    VABufferID    cu_record;
+    /** \brief distortion output of ENC or ENC_PAK.
+     * Each CTB has one distortion data with VAEncFEIDistortionBufferHevc
+     * Buffer size shall not be less than the number of CTBs multiplied by
+     * sizeof(VAEncFEIDistortionBufferHevc).
+     */
+    VABufferID    distortion;
+    /** \brief Qp input buffer. It is valid only when per_block_qp is set to 1.
+     * The data in this buffer correspond to the input source.
+     * One Qp per block block is in raster scan order, each Qp is a signed char (8-bit) value.
+     **/
+    VABufferID    qp;
+    /** \brief MV predictor. It is valid only when mv_predictor_input is set to non-zero.
+     * Each CTB block has one or more pair of motion vectors and the corresponding
+     * reference indexes as defined by VAEncFEIMVPredictorHevc. 32x32 block is in raster scan order.
+     * Buffer size shall not be less than the number of 16x16 blocks multiplied by
+     * sizeof(VAEncFEIMVPredictorHevc). */
+    VABufferID    mv_predictor;
+
+    /** \brief number of MV predictors L0 and L1. the maximum number of motion vector predictor for a 16x16, 32x32 or
+     * 64x64 block is four, it must not be greater than maximum supported MV predictor,
+     **/
+    uint32_t      num_mv_predictors_l0      : 16;
+    uint32_t      num_mv_predictors_l1      : 16;
+
+    /** \brief control parameters */
+    uint32_t      search_path               : 8;
+    uint32_t      len_sp                    : 8;
```

Comment:
what does 'sp' stand for? could you add more comment for this field or use a meaningful name?

Reply:
SP means Search Path, len_sp means Length of Search Path.